src/manager/fevm.rs

```diff
@@ -34,6 +56,168 @@ impl FevmSubnetManager {
     }
 }
 
+#[async_trait]
+impl TopDownCheckpointQuery for FevmSubnetManager {
+    async fn chain_head(&self) -> anyhow::Result<ChainHeadResponse> {
+        self.lotus_client.chain_head().await
+    }
+
+    async fn get_top_down_msgs(
+        &self,
+        subnet_id: &SubnetID,
+        start_epoch: ChainEpoch,
+        end_epoch: ChainEpoch,
+    ) -> anyhow::Result<Vec<CrossMsg>> {
+        self.top_down_msgs(subnet_id, start_epoch, end_epoch).await
+    }
+
+    async fn get_block_hash(&self, height: ChainEpoch) -> anyhow::Result<Vec<u8>> {
+        self.evm_subnet_manager.get_block_hash(height).await
+    }
+}
+
+#[async_trait]
+impl SubnetManager for FevmSubnetManager {
```

Comment:
What has changed here? Why do we need to reimplement this one?

Reply:
Previously it's `EvmSubnetManager`, this is `FevmSubnetManager`.

Reply:
Why do we need both of them? I think we can remove one of them because they're redundant, right?

Reply:
RIght, this is because `evm` and `fevm` are different with existing checkpoint mechanism. Because `fevm` checkpoint bottom up proof requires CID type. But once checkpoint in `fendermint` is merged, we should do a clean up on existing code base. Here is the issue I created to track this: https://github.com/consensus-shipyard/ipc-agent/issues/297